### ISSUE TITLE
Support for headless operation / x-less with finch

### ISSUE THREAD

**Issue by rodneyrod** [elieux/webpidgin#1]
It would be an incredibly good feature to have such an option available so people can remotely access a Pidgin instance on a remote server.

**Comment by elieux** [comment]
I agree, but this is a huge undertaking, so I don't see myself working on this (at least until Pidgin 3 is finally out).

**Comment by rodneyrod** [comment]
What needs to be done? I might be able to help. Even if it was just a text config file it would work.

**Comment by elieux** [comment]
Well, the problem at hand is that I don't understand what specifically you're requesting. I was able to infer the general area of the request and that's why I decided to reply right away, but if we want to discuss this in more detail, I need you to explain your request a bit.

For clarity, I'm noting here that WebPidgin is currently a Pidgin-specific plugin and one needs to run a Pidgin instance to get the web interface and that means installing and running an X server somewhere. I actually do this, but it's definitely not ideal.

One of the simpler remedies would be to make WebPidgin into a pure libpurple plugin, or make a Finch variant of the plugin. The plugin currently contains some amount of Gtk+-dependent code and probably some Pidgin-dependent code (e.g. buddy list traversal, I think), which would need to be changed or rewritten.

An alternative direction would be to create/fork a complete libpurple frontend, i.e. to have "webpurple" sit besides Finch, Pidgin and Adium.

**Comment by rodneyrod** [comment]
Yeah, I had a look at the code and there was a substantial amount of code that pulls things out of Pidgin windows and lists that would pose some difficulty in replacing.

I think though that if this plugin was to be adapted to operate independently of Pidgin, making it depend on Finch or another client probably wouldn't be the best way forward. Making it a pure libpurple plugin or client would probably be the best way forward, I'll have to read up on some documentation relating to that.

**Comment by rodneyrod** [comment]
Also for whatever reason I can't get this to compile on Linux, I get 
make: **\* No rule to make target 'webpidginz.so'. Stop.
when I try to make it in the pidgin source plugin directory.

**Comment by elieux** [comment]
I think you need to configure (and maybe build) pidgin first. (And copy webpidgin*.c to the subdirectory, of course.)

I used this line to avoid dependencies:

```
./configure --disable-screensaver --disable-gtkspell --disable-gstreamer --disable-vv --disable-idn --disable-meanwhile --disable-avahi --disable-nm --disable-perl --disable-nss --disable-gnutls --disable-tcl --disable-mono
```
